setup.py

```diff
@@ -61,6 +61,7 @@
 ]
 
 install_requires = [
+    'celery>=3.1.16,<4.0.0',
```

Comment:
I [investigated a bit](https://github.com/celery/celery/issues/3993#issuecomment-302443875) and I found that there is a compatibility issue with only python2.
Celery is implementing for compatibility reasons [getfullargspec](https://github.com/celery/vine/blob/master/vine/five.py#L349) because doesn't exist in the `inspect` package in py2.
The implementation looks not 100% equivalent to the python 3 `inspect` version and an exception is raised on sphinx.
I propose for now to put `celery<4.0` on docs to be able to run our tests and hope they will fix the [issue](https://github.com/celery/celery/issues/3993).

```python
extras_require = {
    'docs': [
        'Sphinx>=1.5.1,<1.6',
        # FIXME remove when celery#3993 will be resolved!
        'celery<4.0',
    ],
    'tests': tests_require,
}
```